plugins/web/aux/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+find_program(PATCH_EXECUTABLE patch REQUIRED)
+
+if (NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/patched-restinio")
+  file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/patched-restinio")
+endif ()
+
+file(
+  GLOB_RECURSE restinio_sources_relative
+  RELATIVE "${CMAKE_CURRENT_LIST_DIR}/restinio/dev/"
+  CONFIGURE_DEPENDS "${CMAKE_CURRENT_LIST_DIR}/restinio/dev/restinio/*.hpp")
+
+set(restinio_sources_original ${restinio_sources_relative})
+list(TRANSFORM restinio_sources_original
+     PREPEND "${CMAKE_CURRENT_LIST_DIR}/restinio/dev/")
+
+set(restinio_sources_patched ${restinio_sources_relative})
+list(TRANSFORM restinio_sources_patched
+     PREPEND "${CMAKE_CURRENT_BINARY_DIR}/patched-restinio/")
+
+add_custom_command(
+  OUTPUT ${restinio_sources_patched}
+  DEPENDS "${CMAKE_CURRENT_LIST_DIR}/restinio.patch"
+          ${restinio_sources_original}
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/patched-restinio"
+  # Copy the patched restinio library to the binary directory.
+  COMMAND
+    "${CMAKE_COMMAND}" -E copy_directory
+    "${CMAKE_CURRENT_LIST_DIR}/restinio/dev/"
+    "${CMAKE_CURRENT_BINARY_DIR}/patched-restinio"
+  # Apply a patch to the restinio library that makes some changes.
+  COMMAND "${PATCH_EXECUTABLE}" -p1 -i
+          "${CMAKE_CURRENT_LIST_DIR}/restinio.patch")
```

Comment:
Instead of forking, have we considered asking the upstream maintainers whether they'd be willing to accept our contribution?

Reply:
Four notes:
- It wasn't all that complicated to setup. The patching logic is (except for the library name) a carbon copy from the matcher plugin's CMakeLists.txt.
- A fork doesn't maintain itself, so we'd need to actually synchronize it with upstream regularly, which adds additional overhead.
- We're much more likely to get the patch upstreamed if we can point them to this, saying "hey look, we've shown it to work."
- Patching allows us to do this now, and then we can drop the patch as soon as our upstream PR got accepted—which we plan to open once we know that we've got all the changes we need.

Reply:
We agreed to leave it as-is for now since the mechanism already works, as soon as we have tested this in production we want to make an upstream PR and when that is merged we can reset the patch file to be empty again.